components/brave_wallet/browser/internal/hd_key_ed25519.cc

```diff
@@ -5,55 +5,83 @@
 
 #include "brave/components/brave_wallet/browser/internal/hd_key_ed25519.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/check.h"
-#include "base/logging.h"
-#include "base/notreached.h"
-#include "base/strings/strcat.h"
+#include "base/containers/span.h"
+#include "base/containers/span_writer.h"
+#include "base/containers/to_vector.h"
+#include "base/numerics/byte_conversions.h"
 #include "base/strings/string_number_conversions.h"
 #include "brave/third_party/bitcoin-core/src/src/base58.h"
+#include "crypto/hmac.h"
+#include "third_party/boringssl/src/include/openssl/curve25519.h"
 
 namespace brave_wallet {
 namespace {
-constexpr char kMasterNode[] = "m";
+
+inline constexpr char kMasterSecret[] = "ed25519 seed";
+inline constexpr uint32_t kHardenedOffset = 0x80000000;
+
+// Returns concatenation of (private key, pubic key)
+std::array<uint8_t, ED25519_PRIVATE_KEY_LEN> MakeKeyPair(
+    base::span<const uint8_t> private_key,
+    base::span<const uint8_t> public_key) {
```

Comment:
Could we maybe avoid this function by changing

```cxx
  std::array<uint8_t, kEd25519PrivateKeySize> private_key_ = {};
  std::array<uint8_t, kSlip10ChainCodeSize> chain_code_ = {};
  std::array<uint8_t, kEd25519PublicKeySize> public_key_ = {};
```

To

```cxx
struct KeyPair {
  std::array<uint8_t, kEd25519PrivateKeySize> private_key_ = {};
  std::array<uint8_t, kEd25519PublicKeySize> public_key_ = {};
}

KeyPair key_pair_;
```

And then we would be able to do:

```cxx
  std::vector<uint8_t> signature(ED25519_SIGNATURE_LEN);
  if (!ED25519_sign(signature.data(), msg.data(), msg.size(),
                    base::as_span_bytes(key_pair_).data())) {
    return {};
  }
```

Reply:
made storage to be a key pair

Reply:
The problem with a key par is that we don't have the entries named, as opposed with a struct, so it is better to use a struct.

Reply:
To be fair, it may not matter.